platform/posix/transport/src/mbedtls_pkcs11_posix.c

```diff
@@ -0,0 +1,958 @@
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void mbedtlsDebugPrint( void * ctx,
+                               int level,
+                               const char * pFile,
+                               int line,
+                               const char * pStr )
+{
+    /* Unused parameters. */
+    ( void ) ctx;
+    ( void ) pFile;
+    ( void ) line;
+
+    /* Send the debug string to the portable logger. */
+    printf( "mbedTLS: |%d| %s", level, pStr );
+}
+
+/*-----------------------------------------------------------*/
+
+static MbedtlsPkcs11Status_t configureMbedtls( MbedtlsPkcs11Context_t * pMbedtlsPkcs11Context,
+                                               const char * pHostName,
+                                               const MbedtlsPkcs11Credentials_t * pMbedtlsPkcs11Credentials,
+                                               uint32_t recvTimeoutMs )
+{
+    MbedtlsPkcs11Status_t returnStatus = MBEDTLS_PKCS11_SUCCESS;
+    int32_t mbedtlsError = 0;
+
+    assert( pMbedtlsPkcs11Context != NULL );
+    assert( pHostName != NULL );
+    assert( pMbedtlsPkcs11Credentials != NULL );
+    assert( pMbedtlsPkcs11Credentials->pRootCaPath != NULL );
+
+    /* Initialize the MbedTLS context structures. */
+    contextInit( pMbedtlsPkcs11Context );
+    pMbedtlsPkcs11Context->p11Session = pMbedtlsPkcs11Credentials->p11Session;
+
+    mbedtlsError = mbedtls_ssl_config_defaults( &( pMbedtlsPkcs11Context->config ),
+                                                MBEDTLS_SSL_IS_CLIENT,
+                                                MBEDTLS_SSL_TRANSPORT_STREAM,
+                                                MBEDTLS_SSL_PRESET_DEFAULT );
+
+    if( mbedtlsError != 0 )
+    {
+        LogError( ( "Failed to set default SSL configuration: mbedTLSError= %s : %s.",
+                    mbedtlsHighLevelCodeOrDefault( mbedtlsError ),
+                    mbedtlsLowLevelCodeOrDefault( mbedtlsError ) ) );
+
+        /* Per MbedTLS docs, mbedtls_ssl_config_defaults only fails on memory allocation. */
+        returnStatus = MBEDTLS_PKCS11_INSUFFICIENT_MEMORY;
+    }
+    else
+    {
+        /* Set up the certificate security profile, starting from the default value. */
+        pMbedtlsPkcs11Context->certProfile = mbedtls_x509_crt_profile_default;
+
+        /* Set SSL authmode and the RNG context. */
+        mbedtls_ssl_conf_authmode( &( pMbedtlsPkcs11Context->config ), MBEDTLS_SSL_VERIFY_REQUIRED );
+        mbedtls_ssl_conf_rng( &( pMbedtlsPkcs11Context->config ), generateRandomBytes, pMbedtlsPkcs11Context );
+        mbedtls_ssl_conf_cert_profile( &( pMbedtlsPkcs11Context->config ), &( pMbedtlsPkcs11Context->certProfile ) );
+        mbedtls_ssl_conf_read_timeout( &( pMbedtlsPkcs11Context->config ), recvTimeoutMs );
+        mbedtls_ssl_conf_dbg( &pMbedtlsPkcs11Context->config, mbedtlsDebugPrint, NULL );
+        mbedtls_debug_set_threshold( MBEDTLS_DEBUG_LEVEL );
```

Comment:
We don't want to enable logging from mbedTLS by default, instead a new configuration can be exposed as `MBEDTLS_ENABLE_LOGGING` in the header file that is disabled by default. If the user wants to enable logging, they can update the configuration. 
It will be important to add a documentation note in README about the way to enabled mbedTLS logging for debugging TLS connection issues.

Reply:
MBEDTLS_DEBUG_LEVEL is defined in that file, and set to 0, so it is not enabled by default. I'll add docs about it there